### PR TITLE
Allow `node-sass` importers to be passed in options.

### LINKED ISSUE #111: Loading JSON files with sass-loader
I've been using the [sass-json-vars](https://github.com/vigetlabs/sass-json-vars) project to import some configuration that I have shared between JS and Sass. This works by converting the JSON to Sass maps.

I'm trying to switch from the Sass gem to sass-loader and haven't found an easy way to preserve this feature. Do you have any suggestions for how to load JSON files in Sass with sass-loader?

### PR DESCRIPTION
**Pull request by pmowrer**
This allows for [custom importers](https://github.com/sass/node-sass#importer--v200---experimental) to be passed in the loader query.

For example, [node-sass-json-importer](https://github.com/Updater/node-sass-json-importer) could be enabled like so:

```javascript
{
  test: /\.scss$/,
  loaders: [
    "style", 
    "css", 
    "sass?" + JSON.stringify({
      importer: require.resolve('node-sass-json-importer')
    });
  ]
}
```

The above example also resolves #111.

Was hoping to write a test for this, but it seems a bit difficult to fit into the well structured but somewhat rigid test setup. Let me know what you think.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -231,7 +231,18 @@ module.exports = function (content) {
     // indentedSyntax is a boolean flag
     opt.indentedSyntax = Boolean(opt.indentedSyntax);
 
-    opt.importer = getWebpackImporter();
+    // `node-sass`'s importer takes an array of functions. Since `parseQuery` doesn't allow functions,
+    // `opt.importer` has to be a path or array of paths that, once `require`d, return importer functions.
+    opt.importer = []
+        .concat(opt.importer || [])
```
Comment (jhnns) [inline index.js]: This could easier been written like:

```javascript
opt.importer = (opt.importer || [])
    .concat(getWebpackImporter())
```

Comment (pmowrer) [inline index.js]: `opt.importer` may be a string (to mirror `node-sass`s `importer` field which takes a function or array of functions).

**Comment by jhnns** [comment]
Thx for your PR :+1: 

I'm definitely planing to merge this, but it would be good if we'd follow the official recommendation for [programmable objects in query parameters](http://webpack.github.io/docs/how-to-write-a-loader.html#programmable-objects-as-query-option). The sass-loader should not try to `require()` the file, but just read it from the webpack config.

This is also related to #152

**Comment by pmowrer** [comment]
@jhnns Sounds good, thanks for the link (I'm pretty new to Webpack land). I'll make the update soon

**Comment by pmowrer** [comment]
@jhnns Updated per the docs. Let me know what you think.

**Comment by lukasgeiter** [comment]
:+1: for this!

**Comment by jhnns** [comment]
Shipped with `3.1.0`. New preferred way for passing options is now via a property on your webpack config. See https://github.com/jtangelder/sass-loader#sass-options

**Comment by lukasgeiter** [comment]
Awesome, thanks!

**Comment by lencioni** [comment]
Thanks for the work on this. I'm trying to get this set up with node-sass-json-importer and am running into an issue. I'm wondering if someone here can help me out. Here's the configuration I added to my webpack config:

```js
sassLoader: {
  importer: require('node-sass-json-importer'),
}
```

With this configuration sass-loader seems to have a hard time resolving my `@import`s that use `~` and I get messages like the following:

```
ERROR in ./~/css-loader?localIdentName=[path][name]--[local]--[hash:base64:10]!./~/autoprefixer-loader!./~/sass-loader!./app/assets/javascripts/components_loader.js!./app/assets/components/MyComponent/index.scss
Module build failed:
@import '~stylesheets/config';
^
      File to import not found or unreadable: ~stylesheets/config
Parent style sheet: stdin
      in /Users/me/repo/app/assets/components/MyComponent/index.scss (line 1, column 1)
 @ ./app/assets/components/MyComponent/index.scss 4:14-286
```

I found [the bit of code where this is managed in sass-loader](https://github.com/jtangelder/sass-loader/blob/fd06c6c5cafac8792619b0181df2f0fe3faf82e9/index.js#L240-L242) and it seems to add the webpack importer so I'm not really sure why this is failing like this. Any thoughts?

**Comment by lencioni** [comment]
I seem to have the same results with the following configuration, which from what I can tell shouldn't really do much of anything:

```js
sassLoader: {
  importer: (url) => ({ file: url })
},
```

I also tried this, which seemed to work fine:

```js
sassLoader: {
  importer: (_, __, done) => done()
},
```

Do you suppose there might be something else that's odd about my config?

**Comment by jhnns** [comment]
Acccording to the [node-sass docs](https://github.com/sass/node-sass#importer--v200---experimental), you should return `sass.NULL` if your custom importer doesn't want to handle the file. This passes the file request on to the next importer in the array.

**Comment by jhnns** [comment]
If you don't return `sass.NULL` in cases you don't want to resolve the file, all webpack specific resolving won't be applied.

**Comment by lencioni** [comment]
Ah, thanks! It seems that this might be a bug in node-sass-json-importer then (https://github.com/Updater/node-sass-json-importer/blob/f203f2244d157ccce88194d31740d3b54e39128f/src/index.js#L24-L26). I'll move the discussion over there.

**Pull request merged**